index.js

```diff
@@ -1,159 +1,87 @@
-	} catch (err) {
-		cb(null, {
-			title: 'npm',
-			message: 'Not installed. Please install Node.js (which bundles npm) from http://nodejs.org',
-			fail: true
-		});
-		return;
-	}
-
-	execFile(bin, ['--version'], function (err, stdout) {
-		if (err) {
-			cb(err);
-			return;
-		}
+};
 
-		var localVersion = stdout.trim();
+const binaryVersionCheck = bin => {
+	const title = uppercase(bin);
 
-		latestVersion('npm').then(function (version) {
-			var pass = semver.satisfies(localVersion, version);
+	return whichP(bin)
+		.then(() => execa.stdout(bin, ['--version']))
+		.then(localVersion => latestVersion(bin).then(version => {
+			const pass = semver.satisfies(localVersion, version);
 
-			cb(null, {
-				title: 'npm',
-				message: !pass && localVersion + ' is outdated. Please update by running: npm install --global npm',
+			return {
+				title,
+				message: !pass && `${localVersion} is outdated. Please update by running: npm install -g ${bin}`,
 				fail: !pass
-			});
-		});
-	});
-});
-
-checks.push(function yo(cb) {
-	var bin;
+			};
+		}))
+		.catch(err => {
+			if (/not found/.test(err.message)) {
+				return {
+					title,
+					message: `Not installed. Please install it by running: npm install -g ${bin}`,
+					fail: true
+				};
+			}
 
-	try {
-		bin = which.sync('yo');
-	} catch (err) {
-		cb(null, {
-			title: 'yo',
-			message: 'Not installed. Please install it by running: npm install --global yo',
-			fail: true
+			throw err;
 		});
-		return;
-	}
-
-	execFile(bin, ['--version'], function (err, stdout) {
-		if (err) {
-			cb(err);
-			return;
-		}
+};
 
-		var localVersion = stdout.trim();
+const home = {
+	title: process.platform === 'win32' ? '%USERPROFILE' : '$HOME',
+	message: !userHome && 'Environment variable is not set. This is required to know where your home directory is. Follow this guide: https://github.com/sindresorhus/guides/blob/master/set-environment-variables.md',
+	fail: !userHome
+};
 
-		latestVersion('yo').then(function (version) {
-			var pass = semver.satisfies(localVersion, version);
+const node = execa.stdout('node', ['--version']).then(version => {
+	const pass = semver.satisfies(version, '>=4');
 
-			cb(null, {
-				title: 'yo',
-				message: !pass && localVersion + ' is outdated. Please update by running: npm install --global yo',
-				fail: !pass
-			});
-		});
-	});
+	return {
+		title: 'Node.js',
+		message: !pass && `${version} is outdated. Please update: http://nodejs.org`,
+		fail: !pass
+	};
 });
 
-module.exports = function (cb) {
-	var results = [];
+const npm = binaryVersionCheck('npm');
+const yo = binaryVersionCheck('yo');
 
-	eachAsync(checks, function (el, i, next) {
-		el(function (err, result) {
-			if (err) {
-				next(err);
-				return;
-			}
+const checks = [
+	'ruby',
+	'compass',
+	'git'
+].map(x => binaryCheck(x));
 
-			results.push(result);
-			next();
-		});
-	}, function (err) {
-		cb(err, sortOn(results, 'fail'));
-	});
-};
+checks.push(home, node, npm, yo);
```

Comment:
Only this should be resolved and then it's good to merge I guess :). Just review when you find the time.